code/game/objects/items/pneumaticCannon.dm

```diff
@@ -205,6 +205,15 @@
 	var/turf/newtarget = locate(new_x, new_y, starting.z)
 	return newtarget
 
+/obj/item/pneumatic_cannon/handle_atom_del(atom/A)
+	. = ..()
+	if (loadedItems.Remove(A))
+		var/obj/item/I = A
+		loadedWeightClass -= I.w_class
+	if (A == tank)
```

Comment:
else if, since both can't be true

Reply:
not strictly true, but it'd require another bug